File: app/src/main/java/com/komica/reader/adapter/BoardAdapter.java
package com.komica.reader.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.komica.reader.R;
import com.komica.reader.model.Board;
import com.komica.reader.util.FavoritesManager;
import java.util.List;

public class BoardAdapter extends RecyclerView.Adapter<BoardAdapter.BoardViewHolder> {

    private List<Board> boards;
    private BoardCategoryAdapter.OnBoardClickListener onBoardClickListener;
    private OnFavoriteClickListener onFavoriteClickListener;

    public interface OnFavoriteClickListener {
        void onFavoriteClick(Board board);
    }

    public BoardAdapter(List<Board> boards, BoardCategoryAdapter.OnBoardClickListener listener, OnFavoriteClickListener favoriteListener) {
        this.boards = boards;
        this.onBoardClickListener = listener;
        this.onFavoriteClickListener = favoriteListener;
    }

    public void updateData(List<Board> newBoards) {
        this.boards = newBoards;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public BoardViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_board, parent, false);
        return new BoardViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull BoardViewHolder holder, int position) {
        Board board = boards.get(position);
        holder.bind(board);
    }

    @Override
    public int getItemCount() {
        return boards.size();
    }

    class BoardViewHolder extends RecyclerView.ViewHolder {
        private TextView boardName;
        private ImageView favoriteIcon;

        public BoardViewHolder(@NonNull View itemView) {
            super(itemView);
            boardName = itemView.findViewById(R.id.boardName);
            favoriteIcon = itemView.findViewById(R.id.favoriteIcon);
        }

        public void bind(Board board) {
            boardName.setText(board.getName());
            
            FavoritesManager favoritesManager = FavoritesManager.getInstance(itemView.getContext());
            boolean isFavorite = favoritesManager.isFavorite(board.getUrl());
            updateFavoriteIcon(isFavorite);
            
            itemView.setOnClickListener(v -> {
                if (onBoardClickListener != null) {
                    onBoardClickListener.onBoardClick(board);
                }
            });

            favoriteIcon.setOnClickListener(v -> {
                if (onFavoriteClickListener != null) {
                    onFavoriteClickListener.onFavoriteClick(board);
                    // Update UI immediately for better feedback
                    updateFavoriteIcon(favoritesManager.isFavorite(board.getUrl()));
                }
            });
        }
        
        private void updateFavoriteIcon(boolean isFavorite) {
            if (isFavorite) {
                favoriteIcon.setImageResource(R.drawable.ic_star);
            } else {
                favoriteIcon.setImageResource(R.drawable.ic_star_border);
            }
        }
    }
}

File: app/src/main/java/com/komica/reader/adapter/BoardAdapter.kt
package com.komica.reader.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.komica.reader.R
import com.komica.reader.model.Board
import com.komica.reader.util.FavoritesManager

class BoardAdapter(
    private var boards: List<Board>,
    private val onBoardClickListener: BoardCategoryAdapter.OnBoardClickListener?,
    private val onFavoriteClickListener: OnFavoriteClickListener?
) : RecyclerView.Adapter<BoardAdapter.BoardViewHolder>() {

    fun interface OnFavoriteClickListener {
        fun onFavoriteClick(board: Board)
    }

    fun updateData(newBoards: List<Board>) {
        boards = newBoards
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BoardViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_board, parent, false)
        return BoardViewHolder(view)
    }

    override fun onBindViewHolder(holder: BoardViewHolder, position: Int) {
        holder.bind(boards[position])
    }

    override fun getItemCount(): Int = boards.size

    inner class BoardViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val boardName: TextView = itemView.findViewById(R.id.boardName)
        private val favoriteIcon: ImageView = itemView.findViewById(R.id.favoriteIcon)

        fun bind(board: Board) {
            boardName.text = board.name

            val favoritesManager = FavoritesManager.getInstance(itemView.context)
            updateFavoriteIcon(favoritesManager.isFavorite(board.url))

            itemView.setOnClickListener {
                onBoardClickListener?.onBoardClick(board)
            }

            favoriteIcon.setOnClickListener {
                onFavoriteClickListener?.onFavoriteClick(board)
                // 繁體中文註解：點擊最愛後即時更新圖示
                updateFavoriteIcon(favoritesManager.isFavorite(board.url))
            }
        }

        private fun updateFavoriteIcon(isFavorite: Boolean) {
            val icon = if (isFavorite) R.drawable.ic_star else R.drawable.ic_star_border
            favoriteIcon.setImageResource(icon)
        }
    }
}
